### ISSUE TITLE
Variables in each incorrectly idenentified as top scope. 

### ISSUE THREAD

**Issue by dforste** [rodjek/puppet-lint#249]
In the following piece of puppet code 

``` puppet
  if $groups {
    $all_groups.each |$my_group, $group_gid| {
      if !defined( Group[$my_group] ){
        if $my_group in $groups {
          group {$my_group:
            ensure => present,
            gid    => $group_gid,
          }
        }
      }
    }
  }
```

$my_group and $group_gid have the following warning: 
WARNING: top-scope variable being used without an explicit namespace on line ...
